packages/cli/src/lib/remix-version-check.ts

```diff
@@ -0,0 +1,67 @@
+import {createRequire} from 'node:module';
+import {fileURLToPath} from 'node:url';
+import {renderWarning} from '@shopify/cli-kit/node/ui';
+
+export async function checkRemixVersions() {
+  const require = createRequire(import.meta.url);
+  const hydrogenPkgJson = require(fileURLToPath(
+    new URL('../../package.json', import.meta.url),
+  ));
+
+  const requiredVersionInHydrogen = hydrogenPkgJson.dependencies[
+    '@remix-run/dev'
+  ] as string;
+
+  const pkgs = [
+    'dev',
+    'react',
+    'server-runtime',
+    'css-bundle',
+    'node',
+    'express',
+    'eslint-config',
+  ].map((name) => getRemixPackageVersion(require, name));
+
+  const outOfSyncPkgs = pkgs.filter(
+    (pkg) => pkg.version && pkg.version !== requiredVersionInHydrogen,
+  );
+
+  if (outOfSyncPkgs.length === 0) return;
+
+  const items = outOfSyncPkgs.reduce((acc, item) => {
+    if (item.version) {
+      acc.push(`${item.name}@${item.version}`);
+    }
+
+    return acc;
+  }, [] as string[]);
+
+  renderWarning({
+    headline: `The current version of Hydrogen requires Remix @${requiredVersionInHydrogen}. The following packages are out of sync:`,
+    body: [
+      {list: {items}},
+      '\nPlease ensure your Remix packages have the same version and are in sync with Hydrogen.',
+    ],
+  });
+}
+
+type PackageVersion = {
+  name: string;
+  error: null | Error;
+  version: null | string;
+};
+
+function getRemixPackageVersion(require: NodeRequire, name: string) {
+  const pkgName = '@remix-run/' + name;
+  const result: PackageVersion = {name: pkgName, error: null, version: null};
+
+  try {
+    const pkgJsonPath = require.resolve(`${pkgName}/package.json`);
+    const pkgJson = require(pkgJsonPath);
+    result.version = pkgJson.version;
+  } catch (error) {
+    result.error = error as Error;
```

Comment:
Probably fine to just ignore errors?

Reply:
Yeah, didn't end up using them so I guess we can remove this line.